core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3089,6 +3089,21 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
   override def toString: String = "Stream(..)"
 }
 
+class StreamMonad[F[_]] extends Monad[Stream[F, *]] {
+  override def pure[A](x: A): Stream[F, A] = Stream.emit(x)
+
+  override def map[A, B](fa: Stream[F, A])(f: A => B): Stream[F, B] = fa.map(f)
+
+  override def flatMap[A, B](fa: Stream[F, A])(f: A => Stream[F, B]): Stream[F, B] =
+    fa.flatMap(f)
+
+  override def tailRecM[A, B](a: A)(f: A => Stream[F, Either[A, B]]): Stream[F, B] =
+    f(a).flatMap {
+      case Left(a)  => tailRecM(a)(f)
+      case Right(b) => Stream(b)
+    }
```

Comment:
If you do `extends StackSafeMonad[Stream[F, *]]` instead of `Monad` then we don't need this method implementation :)

Reply:
OK!